datacats/environment.py

```diff
@@ -57,6 +62,13 @@ def __init__(self, name, target, datadir, ckan_version=None, port=None,
         self.deploy_target = deploy_target
         self.site_url = site_url
         self.always_prod = always_prod
+        self.profile = UserProfile()
+        if not (self.profile.ssh_private_key is None) and \
+            exists(self.profile.ssh_private_key) and \
+            exists(self.profile.ssh_public_key):
+            self.profile.test_ssh_key(self)
```

Comment:
This shouldn't always be done. Only when using a deploy-related or remote command.